evm-bridging/test/BridgedTopShotMoments.t.sol

```diff
@@ -5,25 +5,74 @@ import {Test} from "forge-std/Test.sol";
+import {IERC721Errors} from "openzeppelin-contracts/contracts/interfaces/draft-IERC6093.sol";
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import {Strings} from "openzeppelin-contracts/contracts/utils/Strings.sol";
+import {CrossVMBridgeERC721FulfillmentUpgradeable} from "../src/lib/CrossVMBridgeERC721FulfillmentUpgradeable.sol";
+import {CrossVMBridgeCallableUpgradeable} from "../src/lib/CrossVMBridgeCallableUpgradeable.sol";
+
+import {IERC721} from "@openzeppelin/contracts/interfaces/IERC721.sol";
+import {IERC721Metadata} from "@openzeppelin/contracts/interfaces/IERC721Metadata.sol";
+import {IERC721Enumerable} from "@openzeppelin/contracts/interfaces/IERC721Enumerable.sol";
+import {ICrossVMBridgeERC721Fulfillment} from "../src/interfaces/ICrossVMBridgeERC721Fulfillment.sol";
+import {ICrossVMBridgeCallable} from "../src/interfaces/ICrossVMBridgeCallable.sol";
+import {ICrossVM} from "../src/interfaces/ICrossVM.sol";
+import {ICreatorToken, ILegacyCreatorToken} from "../src/interfaces/ICreatorToken.sol";
+import {IERC165} from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import {IERC2981} from "@openzeppelin/contracts/interfaces/IERC2981.sol";
+import {IBridgePermissions} from "../src/interfaces/IBridgePermissions.sol";
+
+// Add this minimal ERC721 implementation for testing
+contract UnderlyingERC721 is ERC721, Ownable {
+    constructor(string memory name, string memory symbol) ERC721(name, symbol) Ownable(msg.sender) {}
+
+    function safeMint(address to, uint256 tokenId) public onlyOwner {
+        _safeMint(to, tokenId);
+    }
+}
 
 contract BridgedTopShotMomentsTest is Test {
     address owner;
+    address underlyingNftContractOwner;
+    address underlyingNftContractAddress;
+    address vmBridgeAddress;
+
     string name;
     string symbol;
+    string baseTokenURI;
     string cadenceNFTAddress;
     string cadenceNFTIdentifier;
     string contractURI;
     BridgedTopShotMoments private nftContract;
+    UnderlyingERC721 private underlyingNftContract;
+    uint256[] nftIDs;
 
     // Runs before each test
     function setUp() public {
-        // Set initialization parameters
+        // Set owner
         owner = msg.sender;
+
+        // Deploy underlying NFT contract and mint underlying NFTs to owner
+        underlyingNftContractOwner = address(0x1111);
+        nftIDs = [101, 102, 103];
+        vm.startPrank(underlyingNftContractOwner);
+        underlyingNftContract = new UnderlyingERC721("Underlying NFT", "UNFT");
+        for (uint256 i = 0; i < nftIDs.length; i++) {
+            underlyingNftContract.safeMint(owner, nftIDs[i]);
+        }
+        vm.stopPrank();
+        assertEq(underlyingNftContract.balanceOf(owner), nftIDs.length);
+
+        // Set NFT contract initialization parameters
+        underlyingNftContractAddress = address(underlyingNftContract);
+        vmBridgeAddress = address(0x67890);
```

Comment:
@sisyphusSmiling could you confirm `vmBridgeAddress` should be the address specified in https://github.com/onflow/flow-evm-bridge/blob/main/README.md?plain=1#L24-L25? 🙏

Reply:
Yes, that's correct.